File: packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.java
package pl.leancode.patrol.example;

import androidx.test.platform.app.InstrumentationRegistry;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.junit.runners.Parameterized.Parameters;
import pl.leancode.patrol.PatrolJUnitRunner;
import pl.leancode.patrol.LambdatestPatrolJUnitRunner;


@RunWith(Parameterized.class)
public class MainActivityTest {
    @Parameters(name = "{0}")
    public static Object[] testCases() {
        LambdatestPatrolJUnitRunner instrumentation = (LambdatestPatrolJUnitRunner) InstrumentationRegistry.getInstrumentation();
        instrumentation.setUp(MainActivity.class);
        instrumentation.waitForPatrolAppService();
        return instrumentation.listDartTests();
    }

    public MainActivityTest(String dartTestName) {
        this.dartTestName = dartTestName;
    }

    private final String dartTestName;

    @Test
    public void runDartTest() {
        LambdatestPatrolJUnitRunner instrumentation = (LambdatestPatrolJUnitRunner) InstrumentationRegistry.getInstrumentation();
        instrumentation.runDartTest(dartTestName);
    }
}

File: packages/patrol/example/android/app/src/androidTest/java/pl/leancode/patrol/example/MainActivityTest.kt
package pl.leancode.patrol.example

import android.content.Intent
import androidx.test.rule.ActivityTestRule
import dev.flutter.plugins.integration_test.FlutterTestRunner
import org.junit.Rule
import org.junit.runner.RunWith


class PatrolTestRule : ActivityTestRule<MainActivity>(MainActivity::class.java, true, false) {
    override fun launchActivity(startIntent: Intent?): MainActivity {
//        Logger.i("Starting server loop...")
//
//        val patrolServer = PatrolServer()
//        try {
//            patrolServer.start()
//        } catch (e: Exception) {
//            Logger.e("Failure while running the server: ", e)
//        }
//        // patrolServer.blockUntilShutdown()

        return super.launchActivity(startIntent)
    }
}

@RunWith(FlutterTestRunner::class)
class MainActivityTest {

    @Rule
    var rule = PatrolTestRule()
}
